lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -1033,6 +1046,7 @@ data ApiValidityBound
 data ApiSignTransactionPostData = ApiSignTransactionPostData
     { transaction :: !(ApiT SealedTx)
     , passphrase :: !(ApiT (Passphrase "lenient"))
+    , hexOutput :: !(Maybe Bool)
```

Comment:
![image](https://user-images.githubusercontent.com/1009751/185616745-70553fc5-2dde-4f6c-9399-ff464320ab62.png)

:-)

```suggestion
    , hexOutput :: !(Maybe !Bool)
```

Reply:
I believe the `Just` of the `Maybe` is another arrow if we care about this:
```haskell
import Data.Maybe.Strict (StrictMaybe (..))
:set -XStrictData
data Foo f = Foo { foo :: f Int }
λ> let !a = foo $ Foo $ Just $ undefined
λ> let !a = foo $ Foo $ SJust $ undefined
*** Exception: Prelude.undefined
CallStack (from HasCallStack):
  error, called at libraries/base/GHC/Err.hs:79:14 in base:GHC.Err
  undefined, called at <interactive>:76:30 in interactive:Ghci18
```

Reply:
Actually this might be the same thing as what you said, except that `hexOutput :: !(Maybe !Bool)` isn't allowed?

```
λ> data Foo = Foo { foo :: !(Maybe (!Int)) }

<interactive>:83:34: error:
    • Unexpected strictness annotation: !Int
      strictness annotation cannot appear nested inside a type
    • In the first argument of ‘Maybe’, namely ‘(!Int)’
      In the type ‘(Maybe (!Int))’
      In the definition of data constructor ‘Foo’
```

Reply:
Yes, your example demonstrates my point. `StrictMaybe` is the way.
https://hackage.haskell.org/package/strict-wrapper-0.0.0.0/docs/Data-Strict-Wrapper.html is interesting lib.

Reply:
Alas, I don't think that we need strict fields for our `Api*` data types, because these are intermediate types only used for carrying data across an interface.

If we want to control space usage, it does make sense to maintain [space invariants](https://apfelmus.nfshost.com/blog/2013/08/21-space-invariants.html) on data types that are long-lived (such as `UTxO`), but that is not the case here.